qmk_cli/subcommands/setup.py

```diff
@@ -18,42 +19,44 @@
-@cli.argument('destination', default=os.environ['QMK_HOME'], nargs='?', help='The directory to clone to')
+@cli.argument('-H', '--home', default=Path(os.environ['QMK_HOME']), type=Path, help='The location for QMK Firmware. Defaults to %s' % (os.environ['QMK_HOME'],))
 @cli.argument('fork', default=default_fork, nargs='?', help='The qmk_firmware fork to clone')
 @cli.subcommand('Setup your computer for qmk_firmware.')
 def setup(cli):
-    qmk_firmware = Path(cli.args.destination)
+    """Guide the user through setting up their QMK environment.
+    """
+    clone_prompt = 'Would you like to clone {fg_cyan}%s{fg_reset} to {fg_cyan}%s{fg_reset}?' % (cli.args.fork, shlex.quote(str(cli.args.home)))
+    home_prompt = 'Would you like to set {fg_cyan}%s{fg_reset} as your QMK home?' % (shlex.quote(str(cli.args.home)),)
 
     # Sanity checks
     if cli.args.yes and cli.args.no:
         cli.log.error("Can't use both --yes and --no at the same time.")
         exit(1)
 
     # Check on qmk_firmware, and if it doesn't exist offer to check it out.
-    if (qmk_firmware / 'Makefile').exists():
-        cli.log.info('Found qmk_firmware at %s.', str(qmk_firmware))
+    if (cli.args.home / 'Makefile').exists():
+        cli.log.info('Found qmk_firmware at %s.', str(cli.args.home))
     else:
-        cli.log.error('qmk_firmware not found!')
-        if question('Would you like to clone %s?' % cli.args.fork):
+        cli.log.error('Could not find qmk_firmware!')
+        if question(clone_prompt):
             git_url = '/'.join((cli.config.setup.baseurl, cli.args.fork))
-            clone(git_url, cli.args.destination, cli.config.setup.branch)
+            result = git_clone(git_url, cli.args.home, cli.config.setup.branch)
+            if result != 0:
+                exit(1)
+
+    # Offer to set `user.qmk_home` for them.
+    if cli.args.home != os.environ['QMK_HOME'] and question(home_prompt):
+        cli.config['user']['qmk_home'] = str(cli.args.home.absolute())
+        cli.write_config_option('user', 'qmk_home')
 
     # Run `qmk_firmware/bin/qmk doctor` to check the rest of the environment out
-    if qmk_firmware.exists():
-        qmk_bin = qmk_firmware / 'bin' / 'qmk'
-        doctor_cmd = [sys.executable, str(qmk_bin), 'doctor']
-        if cli.args.yes:
-            doctor_cmd.append('--yes')
-        if cli.args.no:
-            doctor_cmd.append('--no')
-        doctor = subprocess.run(doctor_cmd)
-        if doctor.returncode != 0:
-            cli.log.error('Your build environment is not setup completely.')
-
-            if question('Would you like to run util/qmk_install?'):
-                curdir = os.getcwd()
-                os.chdir(str(qmk_firmware))
-                process = subprocess.run(['util/qmk_install.sh'])
-                os.chdir(curdir)
-                if process.returncode == 0:
-                    cli.log.info('QMK setup complete!')
+    qmk_bin = cli.args.home / 'bin/qmk'
```

Comment:
I guess we could
```suggestion
    qmk_bin = cli.args.home / 'bin' / 'qmk'
```
, but it changes nothing.